yt/frontends/flash/data_structures.py

```diff
@@ -191,25 +191,31 @@ def __init__(
 
         if self.particle_filename is None:
             # try to guess the particle filename
-            try:
-                self._particle_handle = HDF5FileHandler(
-                    filename.replace("plt_cnt", "part")
-                )
-                self.particle_filename = filename.replace("plt_cnt", "part")
-                mylog.info(
-                    "Particle file found: %s", self.particle_filename.split("/")[-1]
-                )
-            except OSError:
+            if "hdf5_plt_cnt" in filename:
```

Comment:
guarding against an unlikely but subtly broken case where this substring is found in the rest of the path (this, and following suggestions assume that `filepath` is defined as `Path(filename)`)
```suggestion
            if "hdf5_plt_cnt" in filepath.name:
```